src/ContractInteract/Anchor.js

```diff
@@ -69,6 +77,129 @@ class Anchor {
     this.anchorStateRoot = this.anchorStateRoot.bind(this);
   }
 
+  /*
+    //Supported Configurations for setup
+    {
+      "remoteChainId": 123456,
+      "deployer": config.deployerAddress,
+      "organization": caOrganization,
+      "coAnchorAddress": caAnchor,
+      "maxStateRoots": maxStateRoots,
+      "organizationOwner": organizationOwner
+    }
+  */
+
+  // TODO: docs
+  static setup(web3, coWeb3, config, txOptions = {}) {
+    Anchor.validateSetupConfig(config);
+
+    const deployParams = Object.assign({}, txOptions);
+    deployParams.from = config.deployer;
+
+    // 1. Get block and stateRoot.
+    const blockHeight = config.blockHeight || 'latest';
+    let initialBlockHeight;
+    let initialStateRoot;
+
+    let promiseChain = coWeb3.eth.getBlock(blockHeight).then((block) => {
+      initialBlockHeight = block.number;
+      initialStateRoot = block.stateRoot;
+    });
+
+    // 2. Deploy the Contract
+    promiseChain = promiseChain.then(() => {
+      return Anchor.deploy(
+        web3,
+        config.remoteChainId,
+        initialBlockHeight,
+        initialStateRoot,
+        config.maxStateRoots,
+        config.organization,
+        deployParams,
+      );
+    });
```

Comment:
```js
    promiseChain = promiseChain.then(() => 
      Anchor.deploy(
        web3,
        config.remoteChainId,
        initialBlockHeight,
        initialStateRoot,
        config.maxStateRoots,
        config.organization,
        deployParams,
      ),
    );
```
? 🤔 

The `.then`s *could* also be chained at the `let promiseChain` above. But it is probably more readable this way. Not sure.